src/config.rs

```diff
@@ -180,12 +210,142 @@ pub fn parse_config() -> Result<Config, Error> {
         ca_root_cert: RootCert::File("./var/run/secrets/istio/root-cert.pem".parse().unwrap()),
         local_xds_config: parse::<PathBuf>(LOCAL_XDS_PATH)?.map(ConfigSource::File),
         xds_on_demand: parse_default(XDS_ON_DEMAND, false)?,
+        proxy_metadata: pc.proxy_metadata,
 
         fake_ca,
         auth: identity::AuthSource::Token(PathBuf::from(r"./var/run/secrets/tokens/istio-token")),
 
-        num_worker_threads: parse_default(ZTUNNEL_WORKER_THREADS, DEFAULT_WORKER_THREADS)?,
+        num_worker_threads: parse_default(
+            ZTUNNEL_WORKER_THREADS,
+            pc.concurrency
+                .unwrap_or(DEFAULT_WORKER_THREADS)
+                .try_into()
+                .expect("concurrency cannot be negative"),
+        )?,
 
         proxy_args: parse_args(),
     })
 }
+
+pub fn default_proxy_config() -> ProxyConfig {
+    ProxyConfig {
+        proxy_admin_port: DEFAULT_ADMIN_PORT,
+        status_port: DEFAULT_STATUS_PORT,
+        concurrency: Some(DEFAULT_WORKER_THREADS),
+        discovery_address: if std::env::var(KUBERNETES_SERVICE_HOST).is_ok() {
+            "https://istiod.istio-system.svc:15012".to_string()
+        } else {
+            "https://localhost:15012".to_string()
+        },
+        termination_drain_duration: Some(DEFAULT_DRAIN_DURATION.try_into().unwrap()),
+        ..Default::default()
+    }
+}
+
+fn construct_proxy_config(mesh_config_path: &str) -> anyhow::Result<ProxyConfig> {
+    let mesh_config: anyhow::Result<serde_yaml::Value> = fs::File::open(mesh_config_path)
+        .map_err(anyhow::Error::new)
+        .and_then(|f| serde_yaml::from_reader(f).map_err(anyhow::Error::new));
+
+    let proxy_config_env: anyhow::Result<serde_yaml::Value> =
+        parse_default::<String>(PROXY_CONFIG, "".to_string())
+            .map_err(anyhow::Error::new)
+            .and_then(|env_str| serde_yaml::from_str(&env_str).map_err(anyhow::Error::new));
+
+    let yaml = match (mesh_config, proxy_config_env) {
+        (Ok(mc), Ok(pc_env)) => {
+            let mut mc = mc
+                .get("defaultConfig")
+                .unwrap_or(&serde_yaml::Value::Null)
+                .clone();
+            merge_yaml(&mut mc, pc_env);
+            Ok(mc)
+        }
+        (Ok(mc), Err(_)) => Ok(mc
+            .get("defaultConfig")
+            .unwrap_or(&serde_yaml::Value::Null)
+            .clone()),
+        (Err(_), Ok(pc_env)) => Ok(pc_env),
+        (Err(mc), Err(pc_env)) => Err(anyhow!("neither parsed: {mc}; {pc_env}")),
```

Comment:
This gives a confusing message: `WARN ztunnel::config: neither parsed: No such file or directory (os error 2); EOF while parsing a value`

There isn't context on what we are trying to do, and "EOF ..." is confusing - maybe we should not have a default for PROXY_CONFIG and just only process it if it is set?